src/game/Spells/Spell.cpp

```diff
@@ -7009,7 +7011,7 @@ SpellCastResult Spell::CheckCast(bool strict)
 
                         if (!boardedOnEachOther)
                             return SPELL_FAILED_NOT_ON_TRANSPORT;
-                    }
+                    }*/
```

Comment:
needs some code analysis. the code does the exact opposite of what it is supposed to do, so I commented it out. but I haven't investigated the mechanism for why it's wrong, yet.